httpx/_utils.py

```diff
@@ -209,6 +210,17 @@ def parse_header_links(value: str) -> typing.List[typing.Dict[str, str]]:
     return links
 
 
+def parse_content_type_charset(content_type: str) -> typing.Optional[str]:
+    # We used to use `cgi.parse_header()` here, but `cgi` became a dead battery.
+    # See: https://peps.python.org/pep-0594/#cgi
+    msg = email.message.Message()
+    msg["content-type"] = content_type
+    charset = msg.get_param("charset")
+    if charset is None:
+        return None
+    return str(charset).strip("'\"")
```

Comment:
does `str` call not seem redundant? given failobj (default to None) is returned in the absence of a param. If param is present, either a string or a 3-tuple. Is later a possibility?

Reply:
True, not sure why I used `str()` here. Also, I think we want `get_content_charset()`, which does an equivalent of `email.utils.collapse_rfc2231_value(msg.get_param("charset"))`, which is what the [get_param() docs](https://docs.python.org/3/library/email.compat32-message.html#email.message.Message.get_param) recommend to deal with this "string or 3-tuple" oddity.